### PR TITLE
Support kafka-exporter synchronous and asynchronous sending

### PR DESCRIPTION
**Pull request by chenhaipeng**
**Description:** In logs and traces scenarios, kafka asynchronous sending has a very good throughput, so I added a sendType feature, which uses sync sending by default

### CODE REVIEW

File: exporter/kafkaexporter/kafka_exporter.go [hunk 1]
```diff
@@ -29,10 +28,193 @@ import (
...
+		k.Run()
+	}
+	k.producer.Input() <- value[0]
+	return nil
+
+}
+func (k *AsyncKafkaProducer) Run() {
+	if k == nil || k.KafkaConfig == nil {
+		return
+	}
```
Comment (MovieStoreGuy) [inline exporter/kafkaexporter/kafka_exporter.go:186]: Testing if `k` is nil here feels like an anti pattern and could lead to issues trying to diagnose problems with the `AsyncKafkaProducer`. It should panic if these aren't correctly set in the worst case (in the best case, return an error on creation).

File: exporter/kafkaexporter/kafka_exporter.go [hunk 2]
```diff
@@ -129,51 +311,51 @@ func (e *kafkaLogsProducer) Close(context.Context) error {
...
-	}
-	c.Producer.Compression = compression
-
-	producer, err := sarama.NewSyncProducer(config.Brokers, c)
-	if err != nil {
-		return nil, err
-	}
-	return producer, nil
-}
+//func newSaramaProducer(config Config) (sarama.SyncProducer, error) {
```
Comment (MovieStoreGuy) [inline exporter/kafkaexporter/kafka_exporter.go:314]: Please don't add commented out code, please just remove it if it isn't needed.

**Review by MovieStoreGuy — CHANGES_REQUESTED** [review]
There is a few things that I would like fixed up here.

File: exporter/kafkaexporter/kafka_exporter.go [hunk 3]
```diff
@@ -29,10 +28,193 @@ import (
...
+	if config.SendType == "" {
+		config.SendType = Sync
+	}
+	switch config.SendType {
+	case Sync:
+		producer = &SyncKafkaProducer{}
+	case Async:
+		producer = &AsyncKafkaProducer{}
+	default:
+		fmt.Println("kafka SendType error please choose sync or async")
```
[the remark this answers is not in the document]
Comment (chenhaipeng) [reply exporter/kafkaexporter/kafka_exporter.go:62]: i will take a look, later

**Comment by chenhaipeng** [comment]
> There is a few things that I would like fixed up here.

hi ,please take a look again ,thanks!

File: exporter/kafkaexporter/kafka_exporter.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (chenhaipeng) [reply exporter/kafkaexporter/kafka_exporter.go]: Hi MovieStoreGuy, I have fixed all problems please look again

**Pull request closed without merging**